src/arrow2/src/array/dyn_ord.rs

```diff
@@ -22,9 +19,9 @@ unsafe fn is_valid<A: Array>(arr: &A, i: usize) -> bool {
 }
 
 #[inline]
-fn compare_with_nulls<A: Array, F: FnOnce() -> Ordering>(
-    left: &A,
-    right: &A,
+fn compare_with_nulls<F: FnOnce() -> Ordering>(
```

Comment:
why the change from static to dynamic dispatch?

Reply:
It's because I don't know the concrete type of null. However I think I might actually go with a different approach that doesn't require this.